hypothesis-python/src/hypothesis/internal/reflection.py

```diff
@@ -318,127 +350,90 @@ def _extract_lambda_source(f):
     # file "<string>", which then returns nonsense to getsource.  Discard it.
     linecache.cache.pop("<string>", None)
 
-    if sig.parameters:
-        if_confused = f"lambda {str(sig)[1:-1]}: <unknown>"
-    else:
-        if_confused = "lambda: <unknown>"
+    # The signature is more informative than the corresponding ast.unparse output,
+    # so add the signature to the unparsed body
+    def format_lambda(body):
+        if sig.parameters:
+            return f"lambda {str(sig)[1:-1]}: {body}"
+        else:
+            return f"lambda: {body}"
+
+    if_confused = format_lambda("<unknown>")
+
     try:
-        source = inspect.getsource(f)
+        source_lines, lineno0 = inspect.findsource(f)
```

Comment:
`inspect.findsource`[isn't documented](https://docs.python.org/3/library/inspect.html) for some reason, but it's publicish enough that I'm happy to rely on it. 

It not being documented might even be an oversight, I might open a cpython issue about it